in_message.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package buffer
+package fuse
```

Comment:
The InMessage and OutMessage classes need to be publicly available to be used along with the MessageProvider interface, so I propose moving these to the fuse package. Please let me know if you have other preferences

Reply:
I’d like to keep the fuse package as small as possible. 

Can you implement it the other way around? Move the MessageProvider interface into the buffer package?